pkg/crc/machine/start.go

```diff
@@ -297,13 +280,38 @@ func (client *client) Start(startConfig StartConfig) (StartResult, error) {
 	}, err
 }
 
+func newMachineConfig(startConfig StartConfig, bundle *bundle.CrcBundleInfo) config.MachineConfig {
+	return config.MachineConfig{
+		Name:          startConfig.Name,
+		BundleName:    filepath.Base(startConfig.BundlePath),
+		CPUs:          startConfig.CPUs,
+		Memory:        startConfig.Memory,
+		DiskPathURL:   fmt.Sprintf("file://%s", filepath.ToSlash(bundle.GetDiskImagePath())),
+		SSHKeyPath:    bundle.GetSSHKeyPath(),
+		KernelCmdLine: bundle.Nodes[0].KernelCmdLine,
+		Initramfs:     bundle.GetInitramfsPath(),
+		Kernel:        bundle.GetKernelPath(),
+	}
+}
+
+func newPostStartConfig(startConfig StartConfig, sshRunner *crcssh.Runner, instanceIP string, bundle *bundle.CrcBundleInfo) services.ServicePostStartConfig {
+	return services.ServicePostStartConfig{
+		Name: startConfig.Name,
+		// TODO: would prefer passing in a more generic type
+		SSHRunner: sshRunner,
+		IP:        instanceIP,
+		// TODO: should be more finegrained
+		BundleMetadata: *bundle,
+	}
+}
+
 func checkDNS(servicePostStartConfig services.ServicePostStartConfig) error {
 	// Check DNS lookup before starting the kubelet
 	if queryOutput, err := dns.CheckCRCLocalDNSReachable(servicePostStartConfig); err != nil {
 		return errors.Wrapf(err, "Failed internal DNS query: %s", queryOutput)
 	}
-	logging.Info("Check internal and public DNS query ...")
 
+	logging.Info("Check internal and public DNS query ...")
```

Comment:
Not the right commit, this belongs to `Add helper for host/VM DNS checks`

Reply:
I dropped this commit, it should be ok for this one.